examples/audio/transformer_asr.py

```diff
@@ -0,0 +1,523 @@
+"""
+Title: Automatic Speech Recognition with Transformer
+Author: [Apoorv Nandan](https://twitter.com/NandanApoorv)
+Date created: 2021/01/13
+Last modified: 2021/01/13
+Description: Implement a complete transformer and train it on sequence to sequence ASR.
+"""
+"""
+## Introduction
+
+Automatic speech recognition (ASR) consists of transcribing audio segments into text.
+A popular way to handle this task is to treat it as a sequence-to-sequence
+problem. The audio can be represented as a sequence of feature vectors,
+and the text can be represented as a sequence of characters, words, or subword tokens.
+
```

Comment:
Suggestion:

```suggestion
ASK tasks can be treat sequence-to-sequence problems, where the
audio can be represented as a sequence of feature vectors
and the text—as a sequence of characters, words, or subword tokens.

```

- In ASR you work on structured sequential data, so ASR systems you build map sequences to sequences. Hence, I'd suggest you say that this is a sequence-to-sequence (referring to the approach developed in 2014) approach "and the audio ..."